app/src/main/java/moe/apex/rule34/DeepLinkActivity.kt

```diff
@@ -0,0 +1,98 @@
+package moe.apex.rule34
+
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.enableEdgeToEdge
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.datastore.preferences.preferencesDataStoreFile
+import coil3.ImageLoader
+import coil3.PlatformContext
+import coil3.SingletonImageLoader
+import coil3.gif.AnimatedImageDecoder
+import coil3.gif.GifDecoder
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import moe.apex.rule34.largeimageview.LargeImageView
+import moe.apex.rule34.preferences.ImageSource
+import moe.apex.rule34.preferences.LocalPreferences
+import moe.apex.rule34.ui.theme.BreadboardTheme
+import moe.apex.rule34.ui.theme.Typography
+
+
+class DeepLinkActivity : SingletonImageLoader.Factory, ComponentActivity() {
+    override fun newImageLoader(context: PlatformContext): ImageLoader {
+        return ImageLoader.Builder(context)
+            .components {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                    add(AnimatedImageDecoder.Factory())
+                } else {
+                    add(GifDecoder.Factory())
+                }
+            }
+            .build()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        enableEdgeToEdge()
+        super.onCreate(savedInstanceState)
+
+        applicationContext.preferencesDataStoreFile("preferences")
+        runBlocking { prefs.handleMigration(applicationContext) }
+        val initialPrefs = runBlocking { prefs.getPreferences.first() }
+
+        setContent {
+            val prefs = prefs.getPreferences.collectAsState(initialPrefs).value
+            CompositionLocalProvider(LocalPreferences provides prefs) {
```

Comment:
Apply the BreadboardTheme here so we don't need to individually wrap the LargeImageView and ImageNotFound (and any other components that we may add in future).